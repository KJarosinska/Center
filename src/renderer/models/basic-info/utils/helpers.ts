import { SimCard } from "Renderer/models/basic-info/interfaces"

export const getActiveNetworkFromSim = (simCards: SimCard[] = []) => {
  const activeSimCard = simCards.filter(simCard => simCard.active)
  if (activeSimCard.length === 0) {
    return "No connection"
  }
  return activeSimCard[0].network
}
