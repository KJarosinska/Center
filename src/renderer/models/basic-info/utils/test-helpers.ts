import { IpcRequest } from "Common/requests/ipc-request.enum"

export const commonCalls = {
  [IpcRequest.GetDeviceInfo]: Promise.resolve({
    name: "Ziemniaczek",
    modelName: "U12300000",
    modelNumber: "A1239999",
    serilaNumber: "a-b-3d",
    osVersion: "0.123v",
    osUpdateDate: "12-12-2003",
  }),
  [IpcRequest.GetNetworkInfo]: Promise.resolve({
    simCards: [
      {
        active: true,
        carrier: "AAAAAAAAAAAA",
        iccid: 1234,
        imei: 5678,
        meid: 8765,
        network: "Y-Mobile",
        networkLevel: 0.5,
        number: 12345678,
        seid: "1234",
        slot: 1,
      },
      {
        active: false,
        carrier: "BBBBBBBBBBBB",
        iccid: 412,
        imei: 42,
        meid: 1410,
        network: "X-Mobile",
        networkLevel: 0.69,
        number: 7001234523,
        seid: "x123",
        slot: 2,
      },
    ],
  }),
  [IpcRequest.GetStorageInfo]: Promise.resolve({
    capacity: 9001,
    available: 99999999999999,
    categories: [
      { label: "music", filesCount: 1233333, size: 999999999 },
      { label: "storage", filesCount: 100000, size: 999999999 },
    ],
  }),
  [IpcRequest.GetBatteryInfo]: Promise.resolve({
    level: 9001,
    charging: false,
    maximumCapacity: 99999,
  }),
  [IpcRequest.GetBackupsInfo]: Promise.resolve({
    backups: [
      {
        createdAt: "20-11-15T07:35:01.562Z20",
        size: 99999,
      },
      {
        createdAt: "20-01-30T07:35:01.562Z20",
        size: 1234567,
      },
    ],
  }),
}
