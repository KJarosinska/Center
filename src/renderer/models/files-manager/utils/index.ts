import { FilesManagerState } from "Renderer/models/files-manager/files-manager.interface"

export const prepareDataForStackedBarChart = (data: FilesManagerState) => {
  return data.memoryData.map(el => {
    const { filesType, filesAmount, occupiedMemory, ...rest } = el
    return {
      value: occupiedMemory,
      ...rest,
    }
  })
}
