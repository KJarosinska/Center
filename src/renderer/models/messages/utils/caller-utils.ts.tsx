import { Author } from "Renderer/models/messages/messages.interface"

const removeDecoratorsFromPhoneNumber = (string = ""): string => {
  return string.split(" ").join("").replace("+", "")
}

export interface CallerSearchParams extends Record<string, string> {
  id: string
  phoneNumber: string
}

export const isCallerMatchingSearchParams = (
  caller: Author,
  params: CallerSearchParams
): boolean => {
  return (
    params.id === caller.id &&
    removeDecoratorsFromPhoneNumber(caller.primaryPhoneNumber) ===
      removeDecoratorsFromPhoneNumber(params.phoneNumber)
  )
}
