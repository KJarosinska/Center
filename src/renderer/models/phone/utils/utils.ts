import Faker from "faker"

// TODO: remove before production
export const generateFakeData = (numberOfContacts: number) => {
  return Array(numberOfContacts)
    .fill(0)
    .map(_ => ({
      id: Faker.random.uuid(),
      firstName: Faker.name.firstName(),
      lastName: Faker.name.lastName(),
      phoneNumber: Faker.phone.phoneNumber(),
      favourite: Faker.random.boolean(),
    }))
}

export const generateSortedStructure = (fakeState: any) => {
  const alphabet = new Array(26)
    .fill(1)
    .map((_, i) => String.fromCharCode(65 + i))

  const sortedContactList = fakeState.sort((a: any, b: any) =>
    a.firstName.localeCompare(b.firstName)
  )

  const generateFakeStructure = () => {
    const fakeStructure = []

    for (const letter of alphabet) {
      fakeStructure.push({
        category: letter,
        contacts: [],
      })
    }

    fakeStructure.unshift({
      category: "Favs",
      contacts: [],
    })

    return fakeStructure
  }

  const placeContactsInStructure = () => {
    const fakeStructure: any = generateFakeStructure()
    fakeStructure.forEach((fakeContact: any) => {
      const contactLetter = fakeContact.category
      const contactList: string[] = fakeContact.contacts
      sortedContactList.forEach((sortedContact: any) => {
        if (
          sortedContact.firstName.charAt(0) === contactLetter &&
          !sortedContact.favourite
        ) {
          contactList.push(sortedContact)
        }
      })
    })

    sortedContactList.forEach((contacts: any) => {
      if (contacts.favourite) {
        fakeStructure[0].contacts.push(contacts)
      }
    })

    return fakeStructure
  }

  return placeContactsInStructure()
}

export const sanitizeContacts = (structure: any) => {
  return structure.filter((el: any) => el.contacts.length > 0)
}

export const filterContacts = (contacts: any, substring: string) => {
  return contacts.map((contactsByLetter: any) => ({
    ...contactsByLetter,
    contacts: contactsByLetter.contacts.filter((contact: any) => {
      const filterableFields = Object.values(contact).filter(value => {
        return typeof value === "string"
      })
      return filterableFields.some((value: any) =>
        value.toLowerCase().includes(substring.toLowerCase())
      )
    }),
  }))
}
