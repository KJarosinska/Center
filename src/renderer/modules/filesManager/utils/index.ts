import {
  FilesManagerData,
  FilesManagerState,
} from "Renderer/models/files-manager/files-manager.interface"

export const memoryDataWithConvertedBytes = (data: FilesManagerState) => {
  return data.memoryData.map((element: FilesManagerData) => ({
    ...element,
    occupiedMemory: element.occupiedMemory / Math.pow(1024, 3),
  }))
}
