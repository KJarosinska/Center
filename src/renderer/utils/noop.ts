export const noop = (...args: unknown[]): void => {
  // no operation here
}
