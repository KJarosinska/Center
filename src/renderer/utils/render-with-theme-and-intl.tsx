import {
  Queries,
  render,
  RenderOptions,
  RenderResult,
} from "@testing-library/react"
import React from "react"
import { IntlProvider } from "react-intl"
import { LANGUAGE } from "Renderer/constants/languages"
import localeEn from "Renderer/locales/main/en-US.json"
import { ThemeProvider } from "styled-components"
import theme from "../styles/theming/theme"

export function renderWithThemeAndIntl(
  ui: React.ReactElement,
  options?: Omit<RenderOptions, "queries">
): RenderResult
export function renderWithThemeAndIntl<Q extends Queries>(
  ui: React.ReactElement,
  options: RenderOptions<Q>
): RenderResult<Q>
export function renderWithThemeAndIntl<Q extends Queries>(
  ui: React.ReactElement,
  options?: RenderOptions<Q> | Omit<RenderOptions, "queries">
) {
  return render<Q>(
    <ThemeProvider theme={theme}>
      <IntlProvider
        defaultLocale={LANGUAGE.default}
        locale={LANGUAGE.default}
        messages={localeEn}
      >
        {ui}
      </IntlProvider>
    </ThemeProvider>,
    // @ts-ignore
    options
  )
}

export function renderWithTheme(
  ui: React.ReactElement,
  options?: Omit<RenderOptions, "queries">
): RenderResult
export function renderWithTheme<Q extends Queries>(
  ui: React.ReactElement,
  options: RenderOptions<Q>
): RenderResult<Q>
export function renderWithTheme<Q extends Queries>(
  ui: React.ReactElement,
  options?: RenderOptions<Q> | Omit<RenderOptions, "queries">
) {
  const originalConsoleError = console.error
  console.error = (...args: any) => {
    if (
      !(
        args[0].startsWith("[React Intl] Cannot format message:") ||
        args[0].startsWith("[React Intl] Missing message:")
      )
    ) {
      originalConsoleError.call(console, ...args)
    }
  }
  return render<Q>(
    <ThemeProvider theme={theme}>
      <IntlProvider locale={LANGUAGE.default}>{ui}</IntlProvider>
    </ThemeProvider>,
    // @ts-ignore
    options
  )
}
