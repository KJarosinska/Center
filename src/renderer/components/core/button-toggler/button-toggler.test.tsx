import { fireEvent } from "@testing-library/dom"
import "@testing-library/jest-dom/extend-expect"
import { wait } from "@testing-library/react"
import "jest-styled-components"
import React from "react"
import ButtonToggler, {
  ButtonTogglerItem,
} from "Renderer/components/core/button-toggler/button-toggler.component"
import {
  singleStateToggler,
  threeStateToggler,
  twoStateToggler,
} from "Renderer/components/core/button-toggler/button-toggler.stories"
import { noop } from "Renderer/utils/noop"
import { renderWithThemeAndIntl } from "Renderer/utils/render-with-theme-and-intl"

const renderButtonToggler = (
  options: typeof singleStateToggler,
  onClick: (label: string) => void = noop
) => {
  const outcome = renderWithThemeAndIntl(
    <ButtonToggler>
      {options.map((label, i) => {
        const onClickHandler = () => onClick(label)
        return (
          <ButtonTogglerItem key={i} label={label} onClick={onClickHandler} />
        )
      })}
    </ButtonToggler>
  )
  return {
    ...outcome,
    getButtons: () => outcome.queryAllByRole("button"),
  }
}

test("matches snapshot", () => {
  const { container } = renderButtonToggler(twoStateToggler)
  expect(container).toMatchSnapshot()
})

test("render single-state toggler properly", () => {
  const { getButtons } = renderButtonToggler(singleStateToggler)
  expect(getButtons()).toHaveLength(singleStateToggler.length)
})

test("render two-state toggler properly", () => {
  const { getButtons } = renderButtonToggler(twoStateToggler)
  expect(getButtons()).toHaveLength(twoStateToggler.length)
})

test("render three-state toggler properly", () => {
  const { getButtons } = renderButtonToggler(threeStateToggler)
  expect(getButtons()).toHaveLength(threeStateToggler.length)
})

test("render buttons labels properly", () => {
  const { getButtons } = renderButtonToggler(twoStateToggler)
  expect(getButtons()[0]).toHaveTextContent(twoStateToggler[0])
  expect(getButtons()[1]).toHaveTextContent(twoStateToggler[1])
})

test("switches active state properly", async () => {
  const onToggle = jest.fn()

  const { getButtons } = renderButtonToggler(threeStateToggler, onToggle)

  const clickOnButton = async (index: number) => {
    fireEvent.click(getButtons()[index])
    await wait(() =>
      expect(onToggle).toHaveBeenCalledWith(threeStateToggler[index])
    )
  }

  await clickOnButton(1)
  await clickOnButton(0)
  await clickOnButton(2)
})
