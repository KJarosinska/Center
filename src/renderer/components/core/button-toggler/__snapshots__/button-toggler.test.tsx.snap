// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
.c3 {
  margin: 0;
  font-size: 1.2rem;
  line-height: 1.2;
  text-transform: uppercase;
  -webkit-letter-spacing: 0.1rem;
  -moz-letter-spacing: 0.1rem;
  -ms-letter-spacing: 0.1rem;
  letter-spacing: 0.1rem;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  background: none;
  -webkit-transition: background 250ms ease-in-out, color 250ms ease-in-out, border 250ms ease-in-out;
  transition: background 250ms ease-in-out, color 250ms ease-in-out, border 250ms ease-in-out;
  cursor: pointer;
  outline: none;
  box-sizing: border-box;
  width: 17rem;
  height: 4rem;
  border-radius: 0.5rem;
  border: 0.1rem solid #3b3f42;
}

.c1 g {
  fill: #000000;
}

.c1:hover {
  border-color: #000000;
}

.c4 {
  white-space: nowrap;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
}

.c2 {
  --radius: 0.5rem;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  z-index: 1;
  pointer-events: all;
}

.c2:hover {
  z-index: 2;
}

.c2:not(:hover) {
  border: 0.1rem solid #cdcecf;
  color: #cdcecf;
}

.c2:not(:hover) g {
  fill: #cdcecf;
}

.c2:not(:first-of-type) {
  border-radius: 0;
  margin-left: -0.1rem;
}

.c2:first-of-type:not(:last-of-type) {
  border-radius: var(--radius) 0 0 var(--radius);
}

.c2:last-of-type:not(:first-of-type) {
  border-radius: 0 var(--radius) var(--radius) 0;
}

<div>
  <section
    class="c0"
  >
    <button
      class="c1 c2"
      size="2"
      type="button"
    >
      <p
        class="c3 c4"
      >
        On
      </p>
    </button>
    <button
      class="c1 c2"
      size="2"
      type="button"
    >
      <p
        class="c3 c4"
      >
        Off
      </p>
    </button>
  </section>
</div>
`;
